src/logger.rs

```diff
@@ -1,27 +1,32 @@
 pub(crate) use lightning::util::logger::Logger;
-use lightning::util::logger::Record;
-use lightning::util::ser::Writer;
 pub(crate) use lightning::{log_error, log_info, log_trace};
 
+use lightning::util::logger::{Level, Record};
+use lightning::util::ser::Writer;
+
 use chrono::Utc;
 
 use std::fs;
 use std::path::Path;
 
 pub(crate) struct FilesystemLogger {
 	file_path: String,
+	level: Level,
 }
 
 impl FilesystemLogger {
-	pub(crate) fn new(file_path: String) -> Self {
+	pub(crate) fn new(file_path: String, level: Level) -> Self {
 		if let Some(parent_dir) = Path::new(&file_path).parent() {
 			fs::create_dir_all(parent_dir).unwrap();
 		}
-		Self { file_path }
+		Self { file_path, level }
 	}
 }
 impl Logger for FilesystemLogger {
 	fn log(&self, record: &Record) {
+		if record.level < self.level {
+			return;
+		}
```

Comment:
Hmm... shouldn't this be enforced at compile time by the log macros?

Reply:
Gotcha... for bindings is there a similar issue in that we construct the log message even if the we ultimately filter?

https://github.com/lightningdevkit/rust-lightning/blob/32eb89474cdc1e400ef89eab76d590d19da4746a/lightning/src/util/logger.rs#L101-L103

Reply:
Ah, right. We aren't exposing the `Logger` interface here.